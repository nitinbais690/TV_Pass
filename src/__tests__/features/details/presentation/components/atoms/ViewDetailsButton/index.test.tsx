import React from 'react';
import { render, toJSON } from '@testing-library/react-native';
import ViewDetailsButton from 'features/details/presentation/components/atoms/ViewDetailButton';
import useAppColors from 'core/presentation/hooks/use-app-colors';

jest.mock('core/presentation/hooks/use-app-colors');

describe('viewDetailsButton', () => {
    test('renders correctly', () => {
        const actionCallback = jest.fn();
        const { container } = render(
            <ViewDetailsButton appColors={useAppColors()} openDetailContent={actionCallback} />,
        );
        expect(toJSON(container)).toMatchSnapshot();
    });
});
