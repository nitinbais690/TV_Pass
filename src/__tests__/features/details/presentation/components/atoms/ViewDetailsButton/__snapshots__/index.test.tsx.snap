// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viewDetailsButton renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    hasTVPreferredFocus={false}
    isTVSelectable={true}
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.07)",
        "marginTop": 23.314285714285713,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 58.285714285714285,
          "justifyContent": "space-between",
          "paddingHorizontal": 23.314285714285713,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 14,
            "fontWeight": "600",
          }
        }
      >
        View more Details
      </Text>
      <SvgMock
        height={24}
        width={24}
      />
    </View>
  </View>
</View>
`;
